network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleClient.java

```diff
@@ -137,9 +137,10 @@ public void registerWithShuffleServer(
       String execId,
       ExecutorShuffleInfo executorInfo) throws IOException {
     checkInit();
-    TransportClient client = clientFactory.createClient(host, port);
+    TransportClient client = clientFactory.createUnmanagedClient(host, port);
```

Comment:
I think it's fine. This executor won't talk to this particular shuffle service after this, so it can close its connection right away.